core/mm/core_mmu.c

```diff
@@ -950,6 +950,50 @@ static void check_sec_nsec_mem_config(void)
 	}
 }
 
+#if defined(CFG_CORE_SEL2_SPMC) && defined(CFG_CORE_PHYS_RELOCATABLE)
+static void collect_device_mem_ranges(struct tee_mmap_region *memory_map,
+				      size_t num_elems, size_t *last)
+{
+	const char *dt_device_match_table = {
+		"arm,ffa-manifest-device-regions",
+	};
+	void *fdt = (void *)manifest_dtb;
+	const char *name = NULL;
+	uint64_t pages_cnt = 0;
+	uint64_t base = 0;
+	int subnode = 0;
+	int node = 0;
+
+	node = fdt_node_offset_by_compatible(fdt, 0, dt_device_match_table);
+	if (node < 0)
+		return;
+
+	fdt_for_each_subnode(subnode, fdt, node) {
+		name = fdt_get_name(fdt, subnode, NULL);
+		if (!name)
+			continue;
+
+		if (dt_getprop_as_number(fdt, subnode, "base-address",
+					 &base)) {
+			EMSG("Mandatory field is missing: base-address");
+			panic();
```

Comment:
I don't think that this should cause a panic, the same below.

Reply:
How about replacing panic with continue，and changing EMSG to IMSG？